grammar.y

```diff
@@ -393,35 +383,86 @@ table_alias:
 ;
 
 join_clause:
-  table_expr JOIN table_expr join_constraint
+  table_expr join_op table_expr join_constraint
   {
     if $4 == nil {
-      $$ = &JoinTableExpr{LeftExpr: $1, JoinOperator: JoinStr, RightExpr: $3}
+      $$ = &JoinTableExpr{LeftExpr: $1, JoinOperator: $2, RightExpr: $3}
     } else {
       $4.LeftExpr = $1
-      $4.JoinOperator = JoinStr
+      $4.JoinOperator = $2
       $4.RightExpr = $3
       $$ = $4
     }
   }
-| join_clause JOIN table_expr join_constraint
+| join_clause join_op table_expr join_constraint
   {
     if $4 == nil {
-      $$ = &JoinTableExpr{LeftExpr: $1, JoinOperator: JoinStr, RightExpr: $3}
+      $$ = &JoinTableExpr{LeftExpr: $1, JoinOperator: $2, RightExpr: $3}
     } else {
       $4.LeftExpr = $1
-      $4.JoinOperator = JoinStr
+      $4.JoinOperator = $2
       $4.RightExpr = $3
       $$ = $4
     }
   }
 ;
 
+join_op:
```

Comment:
and from L410-L460 you have all the grammar rules for a join as shown in the diagram:

<img width="709" alt="Screen Shot 2022-11-21 at 17 27 31" src="https://user-images.githubusercontent.com/1233473/203392372-7a1a5b14-f3d3-4c7d-b457-dd7a43b8104e.png">